DESIGN.md

```diff
@@ -0,0 +1,107 @@
+
+    class Account,Agent,Authority,Channel,Depot,DNS,Identifier,Manners,Storage component
+```
+
+These are only some of the relationships, but it should you a rough idea of some important concepts in the SDK. Starting from the top we have the identifier component that provides an authorisation ticket to the account component, which in turn makes another ticket and passes it to some external account service. Note that this is just an example, it may not use an external service.
+
+Then looking at the 'Storage' component, this would be the session storage. In the default implementation the (file system) data is stored here, but also the authorisation tickets and the cryptographic keys. Then the authority component can redistribute those authorisation tickets if granted by the user, which happens over a channel.
+
+There's some other components such as DNS and "manners" which are somewhat less important, but more on those later.
+
+All of these components are grouped together in a `Program` instance, which will be your main interaction point. It's an artificial environment with the components you've chosen to use. For example, you could have a program that stores session data in memory, or one that stores it in indexedDB.
+
+### Depot
+
+The 'WNFS' file system that holds the user's data consists out of [IPLD](https://ipld.io/) blocks. These blocks make up the content-adressable web, another core idea to the ODD stack. A content address gives us a fingerprint for our data, whenever the data changes, the fingerprint changes. This means that when you provide a specific content address, you will always get back the same data. This means that the storage associated with the file system is tamper-proof, setting us up with an ideal primitive for data syncing.
+
+This component is reponsible for retrieving and storing these IPLD blocks. On the lowest level this means for any given bytes (data) give me back a CID (content address). And vice versa, when I present a CID, give me back the bytes for it.
+
+Looking at this from a higher viewpoint, this unlocks the larger part of data syncing, we can transfer our blocks to a remote source.
+
+Learn more about [content addressing](https://dweb-primer.ipfs.io/avenues-for-access/power-of-content-addressing).
+
+### Account
+
+Most of the time to push the data to a remote endpoint you will need some sort of account system, otherwise anyone could fill up the external storage space which costs money. The user accounts will be represented through authorisation, UCANs, our authorisation tickets.
+
+This ODD SDK program has a UCAN store, a repository of "authorisation tickets" collected from various sources. The account system is one of those sources. It can also access that UCAN store and then use it to authorise various actions.
+
+This may also tie into the other part of data syncing, taking the file system's root CID (aka. data root) and storing it somewhere the other devices can reach it.
```

Comment:
```suggestion
This may also tie into the other part of data syncing: taking the file system's root CID (aka. data root) and storing it somewhere the other devices can reach it.
```